lib/internal/Magento/Framework/App/Bootstrap.php

```diff
@@ -244,29 +244,82 @@ public function createApplication($type, $arguments = [])
      * Runs an application
      *
      * @param \Magento\Framework\AppInterface $application
+     * @param bool $alwaysHandleExceptions Whether to handle exceptions regardless of developer mode setting
      * @return void
      */
-    public function run(AppInterface $application)
+    public function run(AppInterface $application, $alwaysHandleExceptions = true)
```

Comment:
Since some applications require special exception handling, why not use interface polymorphism that is in place already instead of introducing optional argument and IF here? We already have `$application->catchException`.